web/html/src/components/input.js

```diff
@@ -168,214 +169,193 @@ class Form extends React.Component {
-        const invalidHint = isError && this.props.invalidHint;
-        const hint = [this.props.hint, (invalidHint && this.props.hint && <br/>), invalidHint];
-        return (
-            <FormGroup isError={isError}>
-                { this.props.label && <Label name={this.props.label} className={this.props.labelClass} required={this.props.required}/> }
-                <div className={this.props.divClass}>
-                    <input className={"form-control" + (this.props.inputClass ? " " + this.props.inputClass : "")}
-                            type={this.type} name={this.props.name} value={this.props.value}
-                            onChange={this.setValue.bind(this)} disabled={this.props.disabled}
-                            onBlur={this.onBlur.bind(this)} placeholder={this.props.placeholder}/>
-                    { hint &&
-                        <div className="help-block">
-                            {hint}
-                        </div>
-                    }
-                </div>
-            </FormGroup>
-        );
-    }
-}
-
-class Text extends _InputBase {
-    constructor(props) {
-        super(props);
-        this.type = this.props.type || "text";
-    }
-}
-
-class Password extends _InputBase {
-    constructor(props) {
-        super(props);
-        this.type = "password";
-    }
-}
-
-class Check extends _InputBase {
-    constructor(props) {
-        super(props);
-    }
-
-    setChecked(event) {
-        const name = event.target.name;
-        const value = event.target.checked;
-        this.props.onFormChange({
-            name: name,
-            value: value
-        });
-
-        if(this.props.onChange)
-            this.props.onChange(event);
-    }
-
-    render() {
-        const isError = this.state.showErrors && !this.state.isValid;
-        const invalidHint = isError && this.props.invalidHint;
-        const hint = [this.props.hint, (invalidHint && this.props.hint && <br/>), invalidHint];
-        return (
-            <FormGroup isError={isError}>
-                <div className={this.props.divClass}>
-                    <div className="checkbox">
-                        <label>
-                            <input className={this.props.inputClass} name={this.props.name}
-                                    type="checkbox" checked={this.props.value}
-                                    onChange={this.setChecked.bind(this)} onBlur={this.onBlur.bind(this)}
-                                    disabled={this.props.disabled}/>
-                            <span>{this.props.label}</span>
-                        </label>
-                    </div>
-                    { hint &&
-                        <div className="help-block">
-                            {hint}
-                        </div>
-                    }
-                </div>
-            </FormGroup>
-        );
-    }
+function Text(props) {
+  return (
+    <_InputBase
+      name={props.name}
```

Comment:
we could simplify all these props bypassing with:

```javascript
function Text(props) {
  const { specific, ...propsToByPass } = props; 
  return (<_InputBase {...propsToByPass} specific="blahblah" />
}
```

Reply:
cool! I was tired of all of these props :)